manifests/port.pp

```diff
@@ -12,15 +12,16 @@
 #   }
 #
 # @param context A port-context name
-# @param protocol Either tcp or udp. If unset, omits -p flag from semanage.
-# @param port An network port number, like '8514'
-# @param argument An argument for semanage port. Default: "-a"
+# @param protocol Either tcp or udp.
+# @param port An network port number, like '8514', or a range like '1234-4321'
+# @param argument *deprecated*, ignored
 #
 define selinux::port (
   $context,
   $port,
+  $ensure = 'present',
```

Comment:
Are we Puppet4 only now?  If yes I would suggest starting using types now.

```
String $context,
Variant[Integer[1], Array[Integer[1],2,2]] $port,
Enum['present', 'absent'] $ensure = 'present',
Optional[Enum['tcp', 'udp', 'ipv4', 'ipv6']] $protocol = undef,
Optional[String] $argument = undef,
```

`Variant[Integer[1,65535], Array[Integer[1,65535],2,2]]` should allow to pass `port => 1` or `port => [1,34]` for a range. But I really don't know what semanage accepts as `port_range` - I could not find a specification of this.